app/services/locale_normalizer.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class LocaleNormalizer
+  # Normalizes locale string, matching the list of I18n.locales where possible
+  # @param locale [String,Symbol] the locale to normalize
+  # @return [String] the normalized locale
+  def self.normalize_to_i18n(locale)
+    return nil if locale.blank?
+    locale = locale.to_s.gsub("-", "_")
+
+    i18n_pairs.each { |downcased, value| return value if locale.downcase == downcased }
+
+    locale
+  end
+
+  def self.is_same?(locale1, locale2)
+    locale1 = locale1.to_s
+    locale2 = locale2.to_s
```

Comment:
This feels a little odd to me, it seems like the caller should perform the `.to_s` call?

Reply:
We can't rely on the caller to remember to `to_s` all the time. Too many instances of `undefined method `xyz' for an instance of Class` everywhere with ruby.